lib/src/ble/bluez_central.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:bluez/bluez.dart';
+import 'package:ble_ota_app/src/ble/ble_central.dart';
+import 'package:ble_ota_app/src/ble/ble_scanner.dart';
+import 'package:ble_ota_app/src/ble/bluez_scanner.dart';
+
+class BlueZCentral extends BleCentral {
+  BlueZCentral({required this.client}) {
+    () async {
+      await client.connect();
+
+      int attempts = 0;
+      while (attempts < 10 && client.adapters.isEmpty) {
+        await Future.delayed(const Duration(milliseconds: 100));
+        attempts++;
+      }
+
+      if (client.adapters.isEmpty) {
+        _updateCentralStatus(BleCentralStatus.unsupported);
+      } else {
+        _updateCentralStatus(client.adapters.first.powered
+            ? BleCentralStatus.ready
+            : BleCentralStatus.poweredOff);
+      }
+    }.call();
```

Comment:
to avoid this await call workarund from constructor we can use factory function like here:
https://stackoverflow.com/a/59304510

moreover in our case we may want to move this loop logic from constructor.

Reply:
In this case we need to create class immediately, without awaiting on constructor. And wait for status later, in order to not freeze UI. Created  `_init()` method, maybe it will be a solution